src/charm.py

```diff
@@ -409,6 +409,7 @@ def _check_mandatory_config(self) -> None:
                 or self._stored.storage_type not in self.VALID_STORAGE_TYPES
             ):
                 # allow user to change storage type only if initial entry was incorrect
+                assert isinstance(charm_config["storage-type"], str)
                 self._stored.storage_type = charm_config["storage-type"].lower()
```

Comment:
I'm not a fan of using asserts in the middle of the code. Using something like:
```python
self._stored.storage_type = str(charm_config["storage-type"]).lower()
initiators = str(charm_config.get("initiator-dictionary"))
config = str(charm_config.get("multipath-" + section)
```

can fix the lint issues